lab_3_generate_by_ngrams/main.py

```diff
@@ -208,6 +296,20 @@ def build(self) -> int:
         In case of corrupt input arguments or methods used return None,
         1 is returned
         """
+        if not isinstance(self._encoded_corpus, tuple) or\
+                len(self._encoded_corpus) == 0:
+            return 1
+        n_grams = self._extract_n_grams(self._encoded_corpus)
+        if not isinstance(n_grams, tuple) or\
+                len(n_grams) == 0:
+            return 1
+        for n_gram in n_grams:
+            number_of_n_grams = n_grams.count(n_gram)
+            context = tuple(n_gram[:self._n_gram_size - 1])
+            context_count = context.count(n_gram[: -1])
```

Comment:
So... you are counting how many times the first N-1 letters occur in the tuple with the first N-1 letters... looks confusing, does it work?

Reply:
I tried to change it